File: backend/src/models/StudyMaterial.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const studyMaterialSchema = new Schema(
  {
    subjectId: { type: Schema.Types.ObjectId, ref: 'Subject', required: true },
    departmentId: { type: Schema.Types.ObjectId, ref: 'Department', required: true },
    semesterId: { type: Schema.Types.ObjectId, ref: 'Semester', required: true },
    title: { type: String, required: true, trim: true },
    description: { type: String, default: '' },
    // PPT/NOTE are legacy values kept only so any pre-existing documents still deserialize
    // cleanly - new uploads only ever use PDF/VIDEO/DOCUMENT/IMAGE/OTHER (see
    // materialController.MATERIAL_TYPES).
    materialType: { type: String, enum: ['PDF', 'PPT', 'VIDEO', 'NOTE', 'DOCUMENT', 'IMAGE', 'OTHER'], default: 'OTHER' },

    mediaId: { type: Schema.Types.ObjectId, ref: 'Media', required: true },
    fileUrl: { type: String, required: true },
    fileName: { type: String, default: null },
    filePublicId: { type: String, default: null },
    fileResourceType: { type: String, default: null },
    fileSize: { type: Number, default: null },

    uploadedBy: { type: String, ref: 'User', required: true },
  },
  { timestamps: true }
);

module.exports = mongoose.model('StudyMaterial', studyMaterialSchema);

File: app/src/main/java/com/example/uos_lms/core/domain/model/StudyMaterial.java
package com.example.uos_lms.core.domain.model;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@Getter
@Builder(toBuilder = true)
@EqualsAndHashCode
@ToString
public class StudyMaterial {
    private final String id;
    private final String subjectId;
    private final String departmentId;
    private final String semesterId;
    private final String title;
    @Builder.Default
    private final String description = "";
    private final MaterialType materialType;
    private final String fileUrl;
    private final String fileName;
    private final String filePublicId;
    private final String fileResourceType;
    @Builder.Default
    private final long fileSize = 0L;
    private final String uploadedBy;
    private final String uploadedByName;
    @Builder.Default
    private final long uploadedAt = 0L;
}
